app/main/views/suppliers.py

```diff
@@ -422,3 +377,103 @@ def register_interest_in_framework(supplier_id, framework_slug):
             return jsonify(message="Database Error: {0}".format(e)), 400
 
         return jsonify(frameworkInterest=interest_record.serialize()), 201
+
+
+@main.route('/suppliers/<supplier_id>/frameworks/<framework_slug>', methods=['PUT'])
+def register_framework_interest(supplier_id, framework_slug):
+    updater_json = validate_and_return_updater_request()
+
+    framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+
+    supplier = Supplier.query.filter(
+        Supplier.supplier_id == supplier_id
+    ).first()
+
+    if supplier is None:
+        abort(404, "supplier_id '{}' not found".format(supplier_id))
+
+    interest_record = SupplierFramework.query.filter(
+        SupplierFramework.supplier_id == supplier_id,
+        SupplierFramework.framework_id == framework.id
+    ).first()
+
+    if interest_record:
+        return jsonify(frameworkInterest=interest_record.serialize()), 200
+
+    if framework.status != 'open':
+        abort(400, "'{}' framework is not open".format(framework_slug))
+
+    interest_record = SupplierFramework(
+        supplier_id=supplier_id,
+        framework_id=framework.id
+    )
+    audit_event = AuditEvent(
+        audit_type=AuditTypes.register_framework_interest,
+        user=updater_json.get('updated_by'),
+        data={'supplierId': supplier.supplier_id, 'frameworkSlug': framework_slug},
```

Comment:
`supplier.supplier_id ` is when we want the integer from the database record.  The string `supplier_id` comes from the parameter passed in URL and is only used for error messaging (I think?) - seemed to make sense as the error relates to the value passed in the URL.